### DISCUSSION TITLE
Running on a Apache server

### DISCUSSION THREAD

**Discussion by RichardDack** [Prince25/StockAlertBot#36] [General]
Hi, thanks for this i can finally get myself a PS5!

I don't fancy running it locally, but would rather run it on a apace2 web server with proxy pass, i have got it up and running but CORS is causing me a headache. Everything is working fine but the /env and /config files are being blocked by cors. Any advice would be great

 <LocationMatch "/monitor/">
    ProxyPass http://localhost:3250
    ProxyPassReverse http://localhost:3250
   </LocationMatch>

All the best

Rich

**Comment by Prince25** [comment]
Hi Rich. I'm not too familiar with Apache but I'll try to help the best I can. Modern web browsers block local files due to security risks. That is why I couldn't simply use pure JS code to fetch data from the files; I had to use express. What does the error message say exactly? If you're still using my express server for settings, then it should be accepting CORS requests from all origins. Perhaps, you somehow need to do the same with your [apache server](https://stackoverflow.com/questions/38400168/cors-access-control-allow-origin-on-apache-proxy-issue)?

**Reply by RichardDack** [reply]
Thanks for the swift reply!

Yeah im accepting cors headers in the Apache conf, its like just the config and env files are blocked! What's strange is a proxy pass should override the cors and allow it all, as im basically telling Apache to grab localhost and display it. Excuse the ignorance, i believe Express is a web server like Apache? Can i just use Express rather than Apache to serve the webpage? 

Thanks

Rich

**Reply by Prince25** [reply]
That is weird but yes. Yes you can! Here's their [website](https://expressjs.com/) for you to get started. Let me know your progress or if you have further questions!
If my project helps you, please share and consider donating so I can devote more time to this!

**Reply by RichardDack** [reply]
Good Day prince,

 even running the server locally on aws i still get the cors errors
 
![image](https://user-images.githubusercontent.com/47067300/106163195-d1e3af80-6180-11eb-9cc3-5ecb6a36495f.png)

 All the best 
 
 Richard

**Comment by Prince25** [comment]
Hi @RichardDack. Sorry, I wasn't able to reply since I was moving. Just wanted to check in and see if you're still having issues!
